sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/receiver.py

```diff
@@ -5,18 +5,45 @@
 # --------------------------------------------------------------------------
 
 import uuid
+import time
 import logging
-from typing import Optional, Union
+from typing import Optional, Union, TYPE_CHECKING, Callable
 
 from ._decode import decode_payload
 from .link import Link
-from .constants import LinkState, Role
+from .constants import LinkState, Role, LinkDeliverySettleReason
 from .performatives import TransferFrame, DispositionFrame
 from .outcomes import Received, Accepted, Rejected, Released, Modified
 
+if TYPE_CHECKING:
+    from .message import _MessageDelivery
+
 
 _LOGGER = logging.getLogger(__name__)
 
+class PendingDisposition(object):
```

Comment:
Can we fully type-hint any new surface API that gets added?
Might as well start somewhere....